src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/MethodToStateMachineRewriter.cs

```diff
@@ -44,6 +44,8 @@ internal abstract class MethodToStateMachineRewriter : MethodToClassRewriter
         /// </summary>
         protected readonly LocalSymbol cachedState;
 
+        protected readonly LocalSymbol cachedThis;
```

Comment:
Can we add a comment to the purpose of this local.  In particular the cases where it will be non-null?